src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -1004,5 +1005,29 @@ async def setup_venv_pex_process(request: VenvPexProcess) -> Process:
     )
 
 
+@dataclass(frozen=True)
+class VenvPexInfo:
+    """Information captured from `PEX_TOOLS=1 info`."""
+
+    distributions: tuple[str, ...]
+
+
+@rule
+async def extract_venv_pex_info(venv_pex: VenvPex) -> VenvPexInfo:
+    process_result = await Get(
+        ProcessResult,
+        VenvPexProcess(
+            venv_pex,
+            argv=["info"],
+            extra_env={"PEX_TOOLS": "1"},
+            input_digest=venv_pex.digest,
+            description=f"Extract information from {venv_pex.pex.argv0}",
+            level=LogLevel.DEBUG,
+        ),
+    )
```

Comment:
@jsirois is that what you had in mind? This fails with the below when testing on the Flake8 pex, which tells me Flake8's entry point is still being used:

```
pants.engine.process.ProcessExecutionFailure: Process 'Extract information from ./flake8.pex_pex_shim.sh' failed with exit code 1.
stdout:
info:0:1: E902 FileNotFoundError: [Errno 2] No such file or directory: 'info'
```

Reply:
Yeah, you don't want a VenvPexProcess, a PexProcess will do. The PEX file is part of the VenvPex digest. May need to juggle some existing plumbing to get any other info you need. But also you could get more creative. Make a normal PEX but with --include-tools. Run PEX_TOOLS against it. Then create a VenvPex using 1 or 2 scripts and PEX_PATH to link the non venv pex.

Reply:
Or.. make the VenvPex script.smarter. If PEX_TOOLS is in the env, instead of forwarding to the venv's pex script, forward to the PEX file.
https://github.com/pantsbuild/pants/blob/77b99ee30f941268170361f68ea5d96b3a408e39/src/python/pants/backend/python/util_rules/pex.py#L688-L704

Reply:
And the right tool is actually `repository info -v`. You had asked for test coverage of it over here: https://github.com/jsirois/pex/blob/9039b572fed888c488fe0242e80bfae818aa1f99/tests/tools/commands/test_repository.py#L96-L107

Reply:
@Eric-Arellano let me know if you want me to get out of the back seat and take over this WIP. I'm back in town with a keyboard instead of a phone.

Reply:
That would be very appreciated! I tried to figure this out 2 days ago and got lost, including knowing where the original PEX file is located. Switched to other tasks instead.